web/validate.go

```diff
@@ -259,10 +259,18 @@ func runValidator(validator, repopath, commit string, gcl *ginclient.Client) (in
 	return 0, nil
 }
 
-// Root renders the root page of the gin-valid service, which allows the user
+// Root handles the root path of the service. If the user is logged in, it
+// redirects to the user's repository listing. If the user is not logged in, it
+// redirects to the login form.
+func Root(w http.ResponseWriter, r *http.Request) {
+	// Since the /repos path does the same, let's just redirect to that
+	http.Redirect(w, r, "/repos", http.StatusFound)
+}
+
+// PubValidateGet renders the one-time validation form, which allows the user
 // to manually run a validator on a publicly accessible repository, without
 // using a web hook.
-func Root(w http.ResponseWriter, r *http.Request) {
+func PubValidateGet(w http.ResponseWriter, r *http.Request) {
 	if r.Method == http.MethodGet {
```

Comment:
Since the http method is now already checked via the router, could this check be removed?

Reply:
Yes.

Reply:
Should do it for all routes that need it, tbh.